accepted/00000-minion-selective-action-blocking.md

```diff
@@ -0,0 +1,139 @@
+ id               | numeric                
+ label            | character varying(48)  
+ name             | character varying(100) 
+ trigger_snapshot | character(1)                                                                                  
+ unlocked_only    | character(1)
+
+    Table "rhnserverlock"
+  Column   |           Type           
+-----------+--------------------------
+ server_id | numeric                  
+ locker_id | numeric                  
+ reason    | character varying(4000)  
+ created   | timestamp with time zone 
+
+```
+
+If column `unlocked_only` is `Y` the action type is blocked when the system is locked.
+
+Proposed enhancement with locking profiles:
+```
++         Table "rhnLockProfile"
+       Column      |          Type                                                                                         
+ ------------------+------------------------
+  id               | numeric
+  label            | character varying
+ 
++         Table "rhnLockProfileActionTypes"
+       Column      |          Type                                                                                         
+ ------------------+------------------------
+  profile_id       | numeric references rhnLockProfile(id)
+  action_type_id   | numeric references rhnActionType(id)
+  unlocked_only    | character(1)
+ 
+          Table "rhnServerLock"
+   Column    |           Type           
+ ------------+--------------------------
+  server_id  | numeric                  
+  locker_id  | numeric                  
+  reason     | character varying(4000)  
+  created    | timestamp with time zone 
++ profile_id | numeric nullable references rhnLockProfile(id)
+
+```
+The new column `rhnserverlock.profile_id` will reference the locking profile to be used. If it's `NULL` the global configuration from `rhnActionType.unlocked_only` will be used.
+
+## UI and XML-RPC API selective blocking
+
+If a system is locked the UI must check if a certain action is blocked and hide/disable the corresponding UI elements. E.g. if package remove is not allowed then the install button must be disabled in the tab Software -> List
+
+## Implementation phases
+
+The implementation can be split in several phases:
+
+1. Enhance Salt blackout to check metadata
+2. Feature parity with traditional clients - use global locking configuration that's currently defined in table `rhnActionTypes`
+3. Enhance db schema to add locking profiles
+4. Disable UI elements and API calls depending depending on action type when a minion is locked
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+In some cases it's not fined grained enough. Some actions might be forbidden only in certain cases. E.g. in the case of CaaSP, package operations are allowed as long as they don't impact the CaaSP packages.
+
+Such cases need a different solution that could be complementary to system locking, e.g. package locking.
```

Comment:
IIUC the only real solution to that is zypper-level package locking on clients, correct?

Reply:
Right, so far I don't see any other solution.